src/core/layertree/qgslayertreemodellegendnode.cpp

```diff
@@ -419,6 +419,7 @@ QSizeF QgsSymbolLegendNode::drawSymbol( const QgsLegendSettings &settings, ItemC
   QgsRenderContext context;
   context.setScaleFactor( settings.dpi() / 25.4 );
   context.setRendererScale( settings.mapScale() );
+  context.setFlag( QgsRenderContext::Antialiasing, true );
```

Comment:
if ctx->context is available, we should copy the flag from that. (actually, on a wider note, if ctx->context is available we should use that context instead of making a new one here. But I understand if you don't want to handle that in this PR!)

Reply:
@nyalldawson , yeah, won't handle within this PR.